src/parser/mod.rs

```diff
@@ -10045,7 +10055,7 @@ impl<'a> Parser<'a> {
             }
             if self.parse_keywords(&[Keyword::GROUPING, Keyword::SETS]) {
                 self.expect_token(&Token::LParen)?;
-                let result = self.parse_comma_separated(|p| p.parse_tuple(true, true))?;
+                let result = self.parse_comma_separated(|p| p.parse_grouping_set_element())?;
```

Comment:
could we inline the logic vs declaring it as a function? since the logic is only used once and small enough,

```rust
let result = self.parse_comma_separated(|p| {
        if self.peek_token_ref().token == Token::LParen {
            self.parse_tuple(true, true)
        } else {
            Ok(vec![self.parse_expr()?])
        }
})?;
```

Reply:
Sure, done!